### PR TITLE
ci: e2e test splitting

### PR DESCRIPTION
**Pull request by awsluja**
Ready to review
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild_specs/e2e_workflow.yml [hunk 1]
```diff
@@ -97,6 +97,9 @@ batch:
       buildspec: codebuild_specs/run_e2e_tests_linux.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
```
Comment (awsluja) [inline codebuild_specs/e2e_workflow.yml:99]: these are placeholders for local environment runs

File: scripts/split-e2e-tests-codebuild.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+  'src/__tests__/schema-connection-1.test.ts',
+  'src/__tests__/transformer-migrations/searchable-migration.test.ts',
+];
+const TEST_EXCLUSIONS: { l: string[]; w: string[] } = {
+  l: [],
+  w: [
+    'src/__tests__/opensearch-simulator/opensearch-simulator.test.ts',
+    'src/__tests__/storage-simulator/S3server.test.ts',
+    'src/__tests__/amplify-app.test.ts',
+    // failing in parsing JSON strings on powershell
```
Comment (goldbez) [inline scripts/split-e2e-tests-codebuild.ts:52]: nit: can this comment be removed?

File: scripts/split-e2e-tests-codebuild.ts [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+      result.push(tmp);
+    }
+  });
+  return result;
+};
+function main(): void {
+  const configBase: any = loadConfigBase();
+  const baseBuildGraph = configBase.batch['build-graph'];
+
+  const counts = { w: 0, l: 0 };
```
Comment (goldbez) [inline scripts/split-e2e-tests-codebuild.ts:264]: what is this variable for?

File: codebuild_specs/run_e2e_tests_linux.yml [hunk 4]
```diff
@@ -27,16 +25,14 @@ phases:
     commands:
       # you can provide a codebuild source version to use old cache and skip all other jobs :)
       - export NODE_OPTIONS=--max-old-space-size=8096
-      # - export CODEBUILD_SOURCE_VERSION=4610160556171f344780aeb2f0d01b7b20147b5d
       - export AMPLIFY_DIR=$CODEBUILD_SRC_DIR/out
       - export AMPLIFY_PATH=$CODEBUILD_SRC_DIR/out/amplify-pkg-linux-x64
       - echo $AMPLIFY_DIR
       - echo $AMPLIFY_PATH
+      - echo test1
```
Comment (kuhlmanp) [inline codebuild_specs/run_e2e_tests_linux.yml:32]: Is this echo needed?

File: scripts/split-e2e-tests-codebuild.ts (hunk 3 above)
Comment (awsluja) [reply scripts/split-e2e-tests-codebuild.ts:264]: leftover from original test splitting logic, going to remove it; thanks!

File: scripts/split-e2e-tests-codebuild.ts (hunk 2 above)
Comment (awsluja) [reply scripts/split-e2e-tests-codebuild.ts:52]: removing it , thanks

File: codebuild_specs/run_e2e_tests_linux.yml (hunk 4 above)
Comment (awsluja) [reply codebuild_specs/run_e2e_tests_linux.yml:32]: good catch, going to remove now

**Review by kuhlmanp — APPROVED** [review]

**Review by goldbez — APPROVED** [review]

**Pull request merged**